### PR TITLE
#166 - Added a bounding box query test

### PR DESCRIPTION
**Pull request by omerxx**
#166 

Added a bounding box query test based on our tests framework.
It basically tests 3 aspects:

1. Are the resulted markers in accordance with the given latitude/longitude parameters
2. Whether the markers where correctly filtered given `inaccurate` parameter
3. Whether they were filtered correctly given the severity different parameters

### CODE REVIEW

File: tests/test_bounding_box_query.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+                                               start_date=start_date, end_date=end_date,
+                                               fatal=False, severe=True, light=True, inaccurate=False,
+                                               is_thin=False, yield_per=None)
+
+    def tearDown(self):
+        self.query = None
+
+    def test_location_filters(self):
+        for marker in self.query:
+            self.assertEqual(self.query['sw_lat'] < marker.latitude < self.query['ne_lat'], True)
```
Comment (danielhers) [inline tests/test_bounding_box_query.py]: Why not `assertTrue`?

Comment (danielhers) [inline tests/test_bounding_box_query.py]: Should be `<=`, because that's what the filter asks for.

File: tests/__init__.py [hunk 2]
```diff
@@ -0,0 +1 @@
+__author__ = 'Omer'
```
Comment (danielhers) [inline tests/__init__.py]: Is this necessary?

Comment (omerxx) [inline tests/__init__.py]: That allows running a specific commands such as `python -m unittest tests.test_bounding_box_query`
Convenient for testing locally, but I can remove it you don't want it on git

**Comment by danielhers** [comment]
Nice, it's a good test. However, whenever we run it on an empty database, which is the case in Travis CI automated tests, it will all pass vacuously. Can we make it load some sample data before testing it? Perhaps the 2013 data, since it's the sample data that's on Git? In that case, the year should be 2013 rather than the current year.

Regardless, I'll merge this once you handle the line comments.

File: tests/__init__.py (hunk 2 above)
Comment (danielhers) [inline tests/__init__.py]: Great, keep it then.

**Comment by omerxx** [comment]
Not sure what do you mean by making it load the example data.
I changed the default year to 2013 and commented the previous lines for future usability.

Fixed all test calls to `assertTrue` / `assertFalse` where required
Changed to `<=` the bounding box filters

File: tests/test_bounding_box_query.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+from models import Marker  # for Marker.bounding_box_query
+import datetime
+
+# This tests year 2013 accidents as this is the current example git data for testing
+# Once this changes to another year or to the current year's accidents (as should be) un-comment lines 11 & 13
+# and change both 2013 and 2014 to: %s
+
+
+class TestQueryFilters(unittest.TestCase):
+    cyear = str(datetime.datetime.now().strftime("%Y"))
```
Comment (danielhers) [inline tests/test_bounding_box_query.py]: You should comment this out too

**Comment by danielhers** [comment]
I mean to run `process.py` so that the DB isn't empty when the test is run.
It would mean a lot more time to run the tests, though, so I'm not sure it's worth it. Need to think about it.

**Comment by omerxx** [comment]
Done.

**Pull request merged**